### PR TITLE
feat: added Opensearch Serverless module

### LINKED ISSUE #74: [FEATURE] Implement OpenSearch Serverless
There is an OpenSearch module, but that is somewhat costly to keep running.  
Can you add a module for OpenSearch Serverless

### PR DESCRIPTION
**Pull request by srinivasreddych**
*Issue #, if available:*
Fixes #74 

*Description of changes:*
added Opensearch Serverless module

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: f49607fc710d7dc5caf1df764de9e7201327b56b
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a0e2b40d-24ce-482a-8e90-5078e74cb1b2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: ada0ec699787e08dfb9abcacdbff874c9222deb5
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=6851068d-decd-4b82-acc9-6c661269eafc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: config/dev-ssriche.env [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export GLOBAL_CUSTOM_DEPLOYMENT_NAME_IDENTIFIER="ssriche"
-export PRIMARY_ACCOUNT="XXXXXXXXXX"
+export PRIMARY_ACCOUNT="560360184571"
```
Comment (dgraeber) [inline config/dev-ssriche.env:2]: ummm...no

File: manifests/local/networking-modules.yaml [hunk 2]
```diff
@@ -3,4 +3,4 @@ path: modules/network/basic-cdk/
 targetAccount: primary
 parameters:
   - name: internet-accessible
-    value: true
+    value: false
```
Comment (dgraeber) [inline manifests/local/networking-modules.yaml:6]: why?

File: modules/storage/opensearch-serverless/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+### Module Metadata Outputs
+
+- `OpenSearchCollectionEndpoint`: the collection endpoint of Serverless AWS OpenSearch
+- `OpenSearchDashboardEndpoint`: URL of the OpenSearch cluster dashboard
+- `OpenSearchCollectionSecurityGroupId`: the security group associated with Serverless AWS OpenSearch
+
+#### Output Example
+
+```json
+{
```
Comment (dgraeber) [inline modules/storage/opensearch-serverless/README.md:47]: NICE

**Review by dgraeber — COMMENTED** [review]
Was this tested on the latest seedfarmer (3.1.0) just to be sure?

File: manifests/local/networking-modules.yaml (hunk 2 above)
Comment (srinivasreddych) [reply manifests/local/networking-modules.yaml:6]: For testing, i was using a manifest file with and without internet. I can flip this back

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: e21dcc87af4a821ab59faf83abbe403b51a75a08
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=376dc397-6db8-4b3f-a98f-e280e69133d1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/storage/opensearch-serverless/deployspec.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+publishGenericEnvVariables: true
+deploy:
+  phases:
+    install:
+      commands:
+      - npm install -g aws-cdk@2.105.0
+      - pip install -r requirements.txt
+    build:
+      commands:
```
Comment (kukushking) [inline modules/storage/opensearch-serverless/deployspec.yaml:9]: One config suggestion. By default, AOSS will scale up the serverless cluster by 10 OCUs for search and 10 OCUs for index that may add up quite a lot to the bill. We can put a cap on that via:
```
aws opensearchserverless update-account-settings \
    --capacity-limits '{ "maxIndexingCapacityInOCU": 8,"maxSearchCapacityInOCU": 9 }'
    ```

Comment (kukushking) [reply modules/storage/opensearch-serverless/deployspec.yaml:9]: Some info on that [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-scaling.html).

Comment (dgraeber) [reply modules/storage/opensearch-serverless/deployspec.yaml:9]: Shouldn't that be configurable..i.e. cap it only if the customer wants to??

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: a27f34aa1e1b914759b6fe286e2312e0a88430d6
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=253a9322-6329-41f9-8e01-ece8c2ca9bf7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: b44641906bce163626bf1d29d498d7bd6a738ab1
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a552ef91-a6e4-449c-9fe7-81a51a54bd79%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/storage/opensearch-serverless/deployspec.yaml (hunk 4 above)
Comment (srinivasreddych) [reply modules/storage/opensearch-serverless/deployspec.yaml:9]: Documented in the readme. Good point

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: c1bc4d207859e4ca39a9dc3f6142d2b849d892d1
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5dd99ca3-42fb-4e23-842b-af6dbc096dcd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: manifests/local/deployment.yaml [hunk 5]
```diff
@@ -3,14 +3,14 @@ nameGenerator:
...
-toolchainRegion: us-east-1
+toolchainRegion: us-west-2
 groups:
   - name: networking
     path: manifests/local/networking-modules.yaml
   - name: storage
     path: manifests/local/storage-modules.yaml
-  # - name: replication
-  #   path: manifests/local/replicator-modules.yaml
+  - name: replication
```
Comment (dgraeber) [inline manifests/local/deployment.yaml:12]: revert?

File: manifests/local/storage-modules.yaml [hunk 6]
```diff
@@ -1,7 +1,5 @@
 name: buckets
 path: modules/storage/buckets
-targetAccount: primary
```
Comment (dgraeber) [inline manifests/local/storage-modules.yaml:3]: probably want to put the targetAccount back

File: manifests/local/storage-modules.yaml [hunk 7]
```diff
@@ -30,10 +26,24 @@ parameters:
...
+  - name: private-subnet-ids
+    valueFrom:
+      moduleMetadata:
+        group: networking
+        name: basic-networking
+        key: PrivateSubnetIds
+---
 name: efs
 path: modules/storage/efs
-targetAccount: primary
```
Comment (dgraeber) [inline manifests/local/storage-modules.yaml:35]: why remove target account?

File: requirements.txt [hunk 8]
```diff
@@ -1 +1 @@
-seed-farmer~=2.10.0
+seed-farmer~=2.10.0
```
Comment (dgraeber) [inline requirements.txt:1]: should we go 3.1.0 ?

**Review by dgraeber — APPROVED** [review]

File: manifests/local/deployment.yaml (hunk 5 above)
Comment (srinivasreddych) [reply manifests/local/deployment.yaml:12]: Not needed.

File: requirements.txt (hunk 8 above)
Comment (srinivasreddych) [reply requirements.txt:1]: okay... bumping!

File: manifests/local/deployment.yaml (hunk 5 above)
Comment (dgraeber) [reply manifests/local/deployment.yaml]: I don't understand...if the replication module is not needed, why did you add it in this PR?

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 6f20bbbfdffdc8ac624735555051ab6f6366b95e
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=30f376a0-a560-46a5-9ee9-4e087c6bd68e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by dgraeber — APPROVED** [review]

**Pull request merged**